src/App.tsx

```diff
@@ -17,18 +59,30 @@ export const App: React.FC = () => {
             <h1 className="title">Todos:</h1>
 
             <div className="block">
-              <TodoFilter />
+              <TodoFilter
+                inputValue={query}
+                selectedFilter={selectFilter}
+                onSelectStatus={setSelectFilter}
+                onChangeInput={setQuery}
+              />
             </div>
 
             <div className="block">
-              <Loader />
-              <TodoList />
+              {isLoaded
+                ? (
+                  <TodoList
+                    onSelectedTodo={setSelectedTodo}
+                    selectedTodo={selectedTodo}
+                    todos={filteredTodos}
+                    onFind={findTodo}
+                  />
+                ) : <Loader />}
             </div>
           </div>
         </div>
       </div>
 
-      <TodoModal />
+      {selectedTodo && <TodoModal closeModal={setSelectedTodo} todo={selectedTodo} />}
```

Comment:
Either immediately transfer the function that will close the modal, or change the name of `closeModal` prop